pkg/analysis/helpers/extractjsontags/analyzer.go

```diff
@@ -140,10 +145,12 @@ func extractTagInfo(tag *ast.BasicLit) FieldTagInfo {
 
 	return FieldTagInfo{
 		Name:      tagName,
-		OmitEmpty: len(tagValues) == 2 && tagValues[1] == "omitempty",
-		RawValue:  tagValue,
-		Pos:       pos,
-		End:       end,
+		OmitEmpty: (len(tagValues) == 2 && tagValues[1] == omitEmpty) || (len(tagValues) == 3 && (tagValues[1] == omitEmpty || tagValues[2] == omitEmpty)),
+		// Considering, omitzero will always coexist with omitemtpy and it can be either second or third tag.
+		OmitZero: len(tagValues) == 3 && (tagValues[1] == omitZero || tagValues[2] == omitZero),
```

Comment:
This condition is too complexed, then we might embed bug (e.g. some field have multiple tags greater than 3)
How about seeking all tags to check whether the field has specified tag ?

Reply:
Maybe we can turn `tagValues` into a `set`(see https://pkg.go.dev/k8s.io/utils/set), and then use `set.Has` to check if the omitzero/omitempty are present, that looks like it would be a lot simpler than this